### DISCUSSION TITLE
How to understan the parameters in the `application_resnet50()` model?

### DISCUSSION THREAD

**Discussion by danli349** [rstudio/keras#1482] [General]
Hello: 

```{r}
model <- application_resnet50(weights = "imagenet")
summary(model)
```
In the input layer the `Output Shape` is `(None, 224, 224, 3)`
In the `conv1_conv` layer, why the `Output Shape` is `(None, 112, 112, 64)`? 
There is no max pooling layer before it.
Does it omit a (2, 2) max pooling layer, or is the kernel step size 2?
And the kernel size should be (7, 7), right? 
How does the `conv2_block1_1_bn(BatchNormalization)` work layer work? why the parameters are 256?

Thanks a lot

![image](https://github.com/user-attachments/assets/5e8bbf81-a07a-4cd9-8566-0195886182f3)

**Comment by t-kalinowski** [comment]
The first `conv1_conv` layer is configured with `stride=2`. 
https://github.com/keras-team/keras/blob/c8c2c6fc7585a56ca8c8dfd084f5dadad6ad1036/keras/src/applications/resnet.py#L152


One way towards answering the second question is to directly inspect the weights of the layer:

```r
get_layer(model, "conv2_block1_1_bn")$weights
get_layer(model, "conv2_block1_1_bn")$weights |> sapply(\(w) setNames(list(w$shape), w$name)) 
```
```
$gamma
TensorShape([64])

$beta
TensorShape([64])

$moving_mean
TensorShape([64])

$moving_variance
TensorShape([64])
```

Please take a look at the reference page for [`layer_batch_normalization()`](https://keras3.posit.co/reference/layer_batch_normalization.html) for a description of how the different weight tensors are applied.

**Reply by danli349** [reply]
@t-kalinowski Thanks a lot
